pkg/plugins/policies/core/xds/http_filters.go

```diff
@@ -0,0 +1,24 @@
+package xds
+
+import (
+	"errors"
+
+	envoy_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+)
+
+func InsertHTTPFiltersBeforeRouter(manager *envoy_hcm.HttpConnectionManager, newFilters ...*envoy_hcm.HttpFilter) error {
+	if len(newFilters) == 0 {
+		return nil
+	}
+	for i, filter := range manager.HttpFilters {
+		if filter.Name == "envoy.filters.http.router" {
+			// insert new filters before router
+			manager.HttpFilters = append(manager.HttpFilters[:i+len(newFilters)], manager.HttpFilters[i:]...)
```

Comment:
I'm too slow, this is hard to read IMO. Why not do

```
firstHalf := manager.HttpFilters[:i]
secondHalf := manager.HttpFilters[i:]
manager.HttpFilters = append(append(before, newFilters...), secondHalf...)
```

Isn't it also an out of bounds error if `i+len(newFilters)` > `len(manager.HttpFilters)`?